app/audit/models/ChangeReportingMethodNotSavedErrorAuditModel.scala

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package audit.models
+
+import audit.Utilities
+import auth.MtdItUser
+import enums.IncomeSourceJourney.{ForeignProperty, IncomeSourceType, SelfEmployment, UkProperty}
+import play.api.libs.json.{JsValue, Json}
+
+case class ChangeReportingMethodNotSavedErrorAuditModel(incomeSourceType: IncomeSourceType)(implicit user: MtdItUser[_]) extends ExtendedAuditModel {
+
+  override val transactionName: String = enums.TransactionName.ErrorChangeReportingMethodNotSaved
+
+  override val auditType: String = enums.AuditType.ErrorChangeReportingMethodNotSaved
+
+  val journey: String = incomeSourceType match {
```

Comment:
this type conversion seems to be duplicated across a few TxM events;
if this can be moved into ie. /Users/hmrc/devcore/itvc/income-tax-view-change-frontend/app/enums/IncomeSourceJourney/IncomeSourceType.scala

Duplicate code in:
* ObligationsAuditModel
* ManageYourDetailsResponseAuditModel